arch/posix/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+zephyr_cc_option_ifdef(CONFIG_LTO -flto)
+
+zephyr_ld_option_ifdef(CONFIG_COVERAGE
+	-lgcov
+	)
+zephyr_compile_options(
+  -m32
```

Comment:
Will this work on 32-bit Linux? Do we event want to bother with 32-bit Linux? I'd be in favour of dropping support for it, but we'd need a basic check in place.

Reply:
It should. There we are just telling the compiler that the target is i386 (a 32 bit binary). You can also use that if your compile host is a x86, like you can compile from a 32 bit x86 for an ARM target.
The reason to have that switch is to ensure ints and pointers are 32 bits which is the assumption in Zephyr.